packages/react/src/components/stepper/Stepper.tsx

```diff
@@ -0,0 +1,284 @@
+  useEffect(() => {
+    if (stepHeadingRef.current) {
+      stepHeadingRef.current.focus();
+    }
+
+    stepRefs.current[selectedStep - 1].current.scrollIntoView({ behavior: 'smooth', inline: 'center' });
+
+    if (stepperRef.current.scrollLeft > 5) {
+      setShowPreviousButton(true);
+    } else {
+      setShowPreviousButton(false);
+    }
+
+    if (
+      stepperRef.current.scrollWidth - (stepperRef.current.parentNode.clientWidth + stepperRef.current.scrollLeft) >
+      5
+    ) {
+      setShowNextButton(true);
+    } else {
+      setShowNextButton(false);
+    }
+  }, [selectedStep]);
+
+  return (
+    <div lang={language} className={classNames(styles.stepperContainer, customThemeClass)} data-testid={dataTestId}>
+      {showPreviousButton && (
+        <div className={classNames(styles.scrollButton, styles.scrollButtonPrevious)} aria-hidden="true">
+          <button
+            type="button"
+            onClick={() => {
+              const amountToScroll = stepperRef.current.scrollWidth / stepsTotal;
+              stepperRef.current.scrollLeft -= amountToScroll;
+            }}
+            tabIndex={-1}
+          >
+            <IconAngleLeft />
+          </button>
+        </div>
+      )}
+      {showNextButton && (
+        <div className={classNames(styles.scrollButton, styles.scrollButtonNext)} aria-hidden="true">
+          <button
+            type="button"
+            onClick={() => {
+              const amountToScroll = stepperRef.current.scrollWidth / stepsTotal;
+              stepperRef.current.scrollLeft += amountToScroll;
+            }}
+            tabIndex={-1}
+          >
+            <IconAngleRight />
+          </button>
+        </div>
+      )}
+      <div
+        onScroll={(e) => {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+          // @ts-ignore
+          if (e.target.scrollLeft && e.target.scrollLeft > 5) {
+            setShowPreviousButton(true);
+          } else {
+            setShowPreviousButton(false);
+          }
+          // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+          // @ts-ignore
+          if (e.target.scrollWidth - (e.target.parentNode.clientWidth + e.target.scrollLeft) > 5) {
+            setShowNextButton(true);
+          } else {
+            setShowNextButton(false);
+          }
+        }}
+        ref={stepperRef}
+        className={classNames(className, styles.stepper, small && styles.small)}
+      >
+        <div
+          className={styles.line}
+          aria-hidden
+          style={{
+            width: `max( calc(100% - var(--hds-step-width)), calc(${stepsTotal} * var(--hds-step-width) - var(--hds-step-width) ))`,
+          }}
+        >
+          {labels.map((label, index) => {
+            if (index === labels.length - 1) {
+              return null;
+            }
+            return (
+              <div
+                key={`${label}-${index}`} // eslint-disable-line react/no-array-index-key
```

Comment:
Could the lodash.uniqueid fix the index key warning?

Reply:
I wonder if it could cause problems related to server side rendering? We've had some issues before with uniqueId 🤔

Reply:
True, we can leave it like this if it gets too complicated.